src/test/java/de/tum/in/www1/artemis/assessment/ResultServiceIntegrationTest.java

```diff
@@ -367,16 +368,18 @@ else if (i % 4 == 0) {
     }
 
     private void addFeedbacksWithGradingCriteriaToExercise(FileUploadExercise fileUploadExercise) {
-        List<GradingCriterion> gradingCriteria = exerciseUtilService.addGradingInstructionsToExercise(fileUploadExercise);
+        Set<GradingCriterion> gradingCriteria = exerciseUtilService.addGradingInstructionsToExercise(fileUploadExercise);
         gradingCriterionRepository.saveAll(gradingCriteria);
         fileUploadExerciseRepository.save(fileUploadExercise);
 
-        final GradingCriterion criterion1 = getGradingCriterionByTitle(fileUploadExercise, "test title");
-        final GradingCriterion criterion2 = getGradingCriterionByTitle(fileUploadExercise, "test title2");
+        final GradingCriterion criterion1 = GradingCriterionUtil.findGradingCriterionByTitle(fileUploadExercise, "test title");
+        final GradingCriterion criterion2 = GradingCriterionUtil.findGradingCriterionByTitle(fileUploadExercise, "test title2");
 
-        final GradingInstruction instruction1a = criterion1.getStructuredGradingInstructions().get(0);
-        final GradingInstruction instruction1b = criterion1.getStructuredGradingInstructions().get(1);
-        final GradingInstruction instruction2 = criterion2.getStructuredGradingInstructions().get(0);
+        final List<GradingInstruction> gradingInstructions1 = List.copyOf(criterion1.getStructuredGradingInstructions());
+        final GradingInstruction instruction1a = gradingInstructions1.get(0);
+        final GradingInstruction instruction1b = gradingInstructions1.get(1);
+        final List<GradingInstruction> gradingInstructions2 = List.copyOf(criterion2.getStructuredGradingInstructions());
+        final GradingInstruction instruction2 = gradingInstructions2.get(0);
```

Comment:
Is the ordering fixed when using `copyOf` on a Set?

Reply:
No, but it only contains one element anyway. Made that clear in 580ea3a5ea6652446ca0a29ffcb8e5b876c1b847 with an additional assertion on the size.

Reply:
The issue remains for `instruction1a` and `instruction1b` above.

Reply:
→ 48bbbd0ea966d03eea4fc3cbb1ae050f77480f81